tests/kafkatest/services/kafka/templates/kafka.properties

```diff
@@ -52,6 +52,7 @@ quota.consumer.bytes.per.second.overrides={{ quota_config.quota_consumer_bytes_p
 
 security.inter.broker.protocol={{ interbroker_security_protocol }}
 
+ssl.client.auth=required
```

Comment:
I'm surprised we could set this for all uses of `KafkaService`. Aren't some tests completely unaware of anything auth related and therefore likely to fail with this enabled? I'm actually surprised that it seems like only the SSL tests are the ones failing, unless this setting will only take effect when SSL is enabled?